### PR TITLE
The Bark Box vendor for pet items, with new pet items and fun.

### PR DESCRIPTION
**Pull request by ItzLollipop**
## About The Pull Request
Merges most if not all pet fluff items all into their own collective files for people to add onto, which adds two new sets of fluff items, rubber bones, and frisbees. The tennis balls and frisbees not only can be used as fluff items but now usable with special launchers which then can fire them. 
## Why It's Good For The Game
Adds new items for flavor to the server fluff wise, new separate vendor-with a wide selection of items, and contraband/premium items-also allows for more map specific genre's for targeted audiences for more immersive environments. Generally new fun looking stuff is fun for everyone most of the time.
## Changelog
:cl:
add: Chewy bones, frisbees, launchers for tennis balls and frisbees, cargo order for the vendor, machine boards, and main menu backpack loadout slots.
del: Previous ball fluff files
imageadd: Moved ball fluff DMI's into one big fluff DMI's for fluff items, mouth and handheld sprites along with vendor DMI's.
imagedel: Previous ball fluff DMI's.
/:cl:

### CODE REVIEW

File: code/game/objects/items/flufflaunchers.dm [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+	ammo_type = list(/obj/item/ammo_casing/caseless/tennis, /obj/item/ammo_casing/caseless/tennis/red, /obj/item/ammo_casing/caseless/tennis/yellow, /obj/item/ammo_casing/caseless/tennis/green, /obj/item/ammo_casing/caseless/tennis/cyan, /obj/item/ammo_casing/caseless/tennis/blue, /obj/item/ammo_casing/caseless/tennis/purple, /obj/item/ammo_casing/caseless/tennis/rainbow, /obj/item/ammo_casing/caseless/tennis/rainbow/izzy, /obj/item/ammo_casing/caseless/tennis/death)
+	caliber = "ball"
+	max_ammo = 2
+
+/obj/item/projectile/bullet/reusable/ball/classic
+	name = "tennis ball"
+	desc = "BALL!"
+	icon = 'icons/obj/fluff_items.dmi'
+	icon_state = "tennis_classic"
+	damage = 0
```
Comment (Ghommie) [inline code/game/objects/items/flufflaunchers.dm:51]: Save for the icon_state, all these variables are the same for most if not all ball projectiles, so just move them to the `/obj/item/projectile/bullet/reusable/ball` parent type

File: code/game/objects/items/flufflaunchers.dm [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+	ammo_type = list(/obj/item/ammo_casing/caseless/frisbee, /obj/item/ammo_casing/caseless/frisbee/red, /obj/item/ammo_casing/caseless/frisbee/yellow, /obj/item/ammo_casing/caseless/frisbee/green, /obj/item/ammo_casing/caseless/frisbee/cyan, /obj/item/ammo_casing/caseless/frisbee/blue, /obj/item/ammo_casing/caseless/frisbee/purple, /obj/item/ammo_casing/caseless/frisbee/rainbow, /obj/item/ammo_casing/caseless/frisbee/white, /obj/item/ammo_casing/caseless/frisbee/death)
+	caliber = "frisbee"
+	max_ammo = 2
+
+/obj/item/projectile/bullet/reusable/frisbee/classic
+	name = "frisbee"
+	desc = "Catch!"
+	icon = 'icons/obj/fluff_items.dmi'
+	icon_state = "frisbee_classic"
+	damage = 0
```
Comment (Ghommie) [inline code/game/objects/items/flufflaunchers.dm:175]: Same here, for `/obj/item/projectile/bullet/reusable/frisbee`

File: code/game/objects/items/flufflaunchers.dm [hunk 3]
```diff
@@ -0,0 +1,251 @@
+/*		2077 WARCRIMES ARE NOW LEGAL
```
Comment (Ghommie) [inline code/game/objects/items/flufflaunchers.dm:1]: This file should be moved to the `code/modules/projectiles/guns/ballistic` folder.

File: code/game/objects/items/fluff.dm [hunk 4]
```diff
@@ -0,0 +1,296 @@
+/*		FLUFFY FUN
```
Comment (Ghommie) [inline code/game/objects/items/fluff.dm:1]: tennis balls and frisbees should be moved to the `code/modules/projectiles/ammunition/caseless` folder, and the projectiles to the `code/modules/projectiles/projectile/reusable` one.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Tbh, save for special ones, all other tennis balls/frisbees and bones should be of an unique grayscale type (check wirecutters and hyper-eutactic blades for examples).
You really shouldn't follow the same path of whoever added tennis balls in first place, please keep this relatively minor feature slim instead.

File: code/game/objects/items/flufflaunchers.dm (hunk 1 above)
Comment (ItzLollipop) [reply code/game/objects/items/flufflaunchers.dm:51]: Right, so I can just slim it down to the icon_states, gotcha. First time making weapons to be honest.

Comment (ItzLollipop) [reply code/game/objects/items/flufflaunchers.dm:51]: > Tbh, save for special ones, all other tennis balls/frisbees and bones should be of an unique grayscale type (check wirecutters and hyper-eutactic blades for examples).
> You really shouldn't follow the same path of whoever added tennis balls in first place, please keep this relatively minor feature slim instead.

So after a good two hours I've been doing some thinking.
This basically is asking me to change everything I've done, although I'll still do this. I'll probably work on something with the polychromic system, and I'll have to rethink and remake how I've done the weapons. While saying to keep this slim, just possibly doing the polychromic system and reworking how the projectiles are might just expand it further. But that's possibility and I'll have to compare after once I'm done and all that.

Comment (Ghommie) [reply code/game/objects/items/flufflaunchers.dm:51]: Don't do a polychromatic clothing copypaste, just stick with the monochomatic here.

**Comment by Ghommie** [comment]
Stale jank.

**Pull request closed without merging**